src/Features/CSharp/Portable/MoveToResx/CSharpMoveToResxCodeFixProvider.cs

```diff
@@ -0,0 +1,474 @@
+                    .WithTriviaFrom(operation.Literal);
+
+                replacementMap[operation.Literal] = resourceAccess;
+            }
+
+            var newRoot = root.ReplaceNodes(replacementMap.Keys, (original, rewritten) => replacementMap[original]);
+
+            return new DocumentUpdateResult(newRoot);
+        }
+
+        /// <summary>
+        /// Creates the updated solution with both document and resx file changes.
+        /// </summary>
+        private static Task<Solution> CreateUpdatedSolutionAsync(
+            Solution solution,
+            DocumentId documentId,
+            DocumentId resxFileId,
+            SyntaxNode updatedRoot,
+            SourceText? updatedResxText,
+            bool resxRequiresUpdate,
+            CancellationToken cancellationToken)
+        {
+            // Check for cancellation before creating the updated solution
+            cancellationToken.ThrowIfCancellationRequested();
+
+            var newSolution = solution.WithDocumentSyntaxRoot(documentId, updatedRoot);
+
+            if (resxRequiresUpdate && updatedResxText != null)
+            {
+                newSolution = newSolution.WithAdditionalDocumentText(resxFileId, updatedResxText);
+            }
+
+            return Task.FromResult(newSolution);
+        }
+
+        /// <summary>
+        /// Gets the actual resource key to use for a given operation, considering existing resources.
+        /// </summary>
+        public static string GetActualResourceKey(XDocument xdoc, ResourceOperation operation)
+        {
+            // Check if the value already exists in the .resx file
+            var existingDataElement = xdoc.Root?.Elements("data")
+                .FirstOrDefault(e => e.Element("value")?.Value == operation.Value);
+
+            if (existingDataElement is not null)
+            {
+                // Use the existing name
+                var nameAttribute = existingDataElement.Attribute("name");
+                return nameAttribute?.Value ?? operation.Key;
+            }
+
+            return operation.Key;
+        }
+
+        /// <summary>
+        /// Result of updating a resx file.
+        /// </summary>
+        private readonly record struct ResxUpdateResult(bool IsSuccess, SourceText? UpdatedText, bool RequiresUpdate)
+        {
+            public static ResxUpdateResult Success(SourceText? updatedText, bool requiresUpdate)
+                => new(true, updatedText, requiresUpdate);
+
+            public static ResxUpdateResult Failed()
+                => new(false, null, false);
+        }
+
+        /// <summary>
+        /// Result of updating a document.
+        /// </summary>
+        private readonly record struct DocumentUpdateResult(SyntaxNode UpdatedRoot);
+    }
+
+    private class CSharpMoveToResxFixAllProvider : FixAllProvider
+    {
+        public override IEnumerable<FixAllScope> GetSupportedFixAllScopes()
+            => new[] { FixAllScope.Document };
```

Comment:
```suggestion
            => [FixAllScope.Document];
```

Why only fix all in document?